### PR TITLE
Re-adds Hunter Silence

### PR DESCRIPTION
**Pull request by KrabSpider**
## About The Pull Request
Re-adds silence to Hunter's toolkit; Silence is an ability that mutes targets for 10 seconds while also impairing their hearing and vision for 7 seconds. Inflicts 1 stagger stack on targets upon use.
Also 
Cooldown is 30 seconds (3 seconds on whiff)
Plasma cost is 50.
Hunter's Mark Targets suffer a 50% longer duration, or about 15 seconds. 
## Why It's Good For The Game
It got removed in https://github.com/tgstation/TerraGov-Marine-Corps/pull/10076.
Silence is a (situational) ability that gives Hunter's some way to prevent or delay Marine communication, while also providing even more incentive for non-primo Hunters to make use of Psychic Trace and Hunter's Mark. 

## Changelog
:cl:
add: Re-adds Silence, a Hunter-exclusive ability that mutes and deafens hostile targets the Hunter has line of sight to in a 5x5 area for 10 seconds. Hunter's Mark targets suffer a 50% longer duration.
/:cl:

### CODE REVIEW

**Comment by shelteredfreak** [comment]
May the gods (maintainers) smile brightly upon your soul kind stranger, Hunter will finally be able to scare the shit out of marines like the days of old.

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 1]
```diff
@@ -489,3 +489,87 @@
...
+// *********** Silence
+// ***************************************
+/datum/action/xeno_action/activable/silence
+	name = "Silence"
+	action_icon_state = "silence"
+	desc = "Impairs the ability of hostile living creatures we can see in a 5x5 area. Targets will be unable to speak and hear for 10 seconds, or 15 seconds if they're your Hunter Mark target."
+	ability_name = "silence"
+	plasma_cost = 50
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_SILENCE,)
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:503]:
```suggestion
	keybinding_signals = list(
		KEYBINDING_NORMAL = COMSIG_XENOABILITY_SILENCE,
	)
```

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 2]
```diff
@@ -489,3 +489,87 @@
...
+	. = ..()
+	if(!.)
+		return
+
+	var/mob/living/carbon/xenomorph/impairer = owner //Type cast this for on_fire
+
+	var/distance = get_dist(impairer, A)
+	if(distance > HUNTER_SILENCE_RANGE)
+		if(!silent)
+			to_chat(impairer, "<span class='xenodanger'>The target location is too far! We must be [distance - HUNTER_SILENCE_RANGE] tiles closer!</spam>")
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:516]:
```suggestion
			to_chat(impairer, span_xenodanger("The target location is too far! We must be [distance - HUNTER_SILENCE_RANGE] tiles closer!"))
```

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 3]
```diff
@@ -489,3 +489,87 @@
...
+
+
+
+/datum/action/xeno_action/activable/silence/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+
+	X.face_atom(A)
+
+	var/victim_count
+	for(var/mob/living/target AS in hearers(HUNTER_SILENCE_AOE, A))
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:534]: Refactor this to use `cheap_get_humans_near`

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 4]
```diff
@@ -489,3 +489,87 @@
...
+				continue
+
+		var/silence_multiplier = 1
+		var/datum/action/xeno_action/activable/hunter_mark/mark_action = X.actions_by_path[/datum/action/xeno_action/activable/hunter_mark]
+		if(mark_action?.marked_target == target) //Double debuff stacks for the marked target
+			silence_multiplier = HUNTER_SILENCE_MULTIPLIER
+		to_chat(target, span_danger("Your mind convulses at the touch of something ominous as the world seems to blur, your voice dies in your throat, and everything falls silent!") ) //Notify privately
+		target.playsound_local(target, 'sound/effects/ghost.ogg', 25, 0, 1)
+		target.adjust_stagger(HUNTER_SILENCE_STAGGER_STACKS * silence_multiplier)
+		target.adjust_blurriness(HUNTER_SILENCE_SENSORY_STACKS * silence_multiplier)
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:553]: Didn't say in the PR desc that it staggers and blurs eyes, stinky.

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm [hunk 5]
```diff
@@ -489,3 +489,87 @@
...
+		target.adjust_ear_damage(HUNTER_SILENCE_SENSORY_STACKS * silence_multiplier, HUNTER_SILENCE_SENSORY_STACKS * silence_multiplier)
+		target.apply_status_effect(/datum/status_effect/mute, HUNTER_SILENCE_MUTE_DURATION * silence_multiplier)
+		victim_count++
+
+	if(!victim_count)
+		to_chat(X, "<span class='xenodanger'>We were unable to violate the minds of any victims.")
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:559]:
```suggestion
		to_chat(X, span_xenodanger("We were unable to violate the minds of any victims."))
```

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm (hunk 3 above)
Comment (KrabSpider) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:535]: Done.

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm (hunk 4 above)
Comment (KrabSpider) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:554]: Updated the PR body.

File: code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm (hunk 3 above)
Comment (ivanmixo) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm:535]: No it isn't done. Stop resolving reviews you haven't resolved.

Comment (KrabSpider) [reply code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm]: I forgot to push the commit before I had left for work. My mistake.

**Review by ivanmixo — APPROVED** [review]

**Pull request merged**